x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/AppliedQueryRules.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.rules;
+
+import org.elasticsearch.xpack.searchbusinessrules.PinnedQueryBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.searchbusinessrules.PinnedQueryBuilder.Item;
+
+public class AppliedQueryRules {
+
+    private final List<String> pinnedIds;
+    private final List<Item> pinnedDocs;
+
+    public AppliedQueryRules() {
+        this(new ArrayList<>(0), new ArrayList<>(0));
+    }
+
+    public AppliedQueryRules(List<String> pinnedIds, List<Item> pinnedDocs) {
+        this.pinnedIds = pinnedIds;
+        this.pinnedDocs = pinnedDocs;
+    }
+
+    public List<String> pinnedIds() {
+        return pinnedIds;
+    }
+
+    public List<Item> pinnedDocs() {
+        return pinnedDocs;
+    }
+
+    public void applyRule(QueryRule queryRule, Map<String, Object> matchCriteria) {
+        if (QueryRuleValidator.validRule(queryRule)) {
+            if (queryRule.type() == QueryRule.QueryRuleType.PINNED) {
+                applyPinnedRule(queryRule, matchCriteria);
```

Comment:
I guess this is fine for now since we have a single type of rule, but in the future could we delegate applying the rule to the `QueryRule` object? Otherwise every time we add a new rule type, we need to add a check like this and another `applyNewTypeRule` method here

Reply:
I can do some refactoring to simplify this but maybe in a followup PR?

Reply:
I did a bit of refactoring here but stopped short of creating a class hierarchy or anything like that since we only have one type of rule right now. CC: @carlosdelest

Reply:
Perfectly fine and it might not be needed until we add another query rule type